ddtrace/contrib/aiohttp/middlewares.py

```diff
@@ -108,7 +108,7 @@ def on_prepare(request, response):
     if trace_query_string:
         request_span.set_tag(http.QUERY_STRING, request.query_string)
 
-    url = request.url.with_query(None)
+    url = str(request.url.with_query(None))
```

Comment:
is this unnecessarily expensive for us?

that is, converting a url object to a string, to then parse it into url parts, to then put it back together?

Especially if we can do `.with_query(None)` here, seems like this is a place where we are doing unnecessary work?

Reply:
Yes, this seems to be doing quite some unnecessary work. Perhaps a smarter thing to do is to ditch `urllib` and write an ad-hoc function that removes the qs from a url.

Reply:
Worth noting in Python 3.8 on my macbook with `http://example.com/search?q=test+query#frag?ment` that @P403n1x87's sanitize implementation runs in about **`415 ns`** whereas the urllib code ran in **`4.13 µs`**

Reply:
Ah I suppose that I could drop the call to `with_query(None)` here?

Reply:
Looks like it!

Reply:
well, if there is a built-in API in order to drop the query string, why do we need to call our function at all?

Reply:
as far as I understand, that works because `request.url` is a yarl `URL` object, and more generally the change now calls `strip_query_string` on everything that's passed to `set_http_meta`, so there's no way of avoiding it when going through this helper.

Reply:
ah, right!